### PR TITLE
Colin/withdrawal delay

### PR DESCRIPTION
**Pull request by colin-axner**
Addresses #42 
**Name Changes**
Priority: A 256 bit value where the left 128 bits correspond to when the utxo is exitable. If the utxo is older than 1 week, this is the block.timestamp + 1 week, otherwise it is utxo's creation time + 2 weeks. The right 128 bits represent the utxo's position. The priority is only used for storage in the priority queue.

Position: 256 bit unique value representing the position of a utxo on the plasma sidechain. This is represented by blockFactor \* blockNum + txIndex \* txIndexFactor + oIndex. This is as the key value when mapping to exits. This value users will be aware of.

I was forced to change the structure of startExitTransaction slightly because I was receiving a stack too deep error since too many local variables were being used

### CODE REVIEW

File: contracts/rootchain/RootChain.sol [hunk 1]
```diff
@@ -161,36 +162,46 @@ contract RootChain is Ownable {
...
             state: ExitState.Pending
         });
 
-        emit StartedTransactionExit(priority, msg.sender, amount);
+        emit StartedTransactionExit(position, msg.sender, amount);
     }
 
+    function validateProofAndSignatures(uint256[3] txPos, bytes txBytes, bytes proof, bytes sigs, bytes confirmSignatures, RLPReader.RLPItem[] txList)
+        private 
+        view
```
[the remark this answers is not in the document]
Comment (colin-axner) [reply contracts/rootchain/RootChain.sol]: use of childChain[txPos[0]].root prevents this change

File: test/rootchain/transactions.js [hunk 2]
```diff
@@ -231,4 +232,112 @@ contract('[RootChain] Transactions', async (accounts) => {
...
+        
+        // increase time slightly, so exit by accounts[2] has better priority than accounts[1]
+        fastForward(10);
+
+        // start exit for accounts[1] utxo
+        await rootchain.startTransactionExit([blockNum2, 0, 0], 
+            toHex(txBytes2), toHex(proof2), toHex(sigs2), toHex(confirmSigs2), {from: accounts[1], value: minExitBond});
+        
+        // Fast Forward ~5 days
+        fastForward(432000);
```
Comment (AdityaSripal) [inline test/rootchain/transactions.js]: Could this be calculated explicitly in code for readability?

**Comment by AdityaSripal** [comment]
Also, are confirm sigs NEEDED if the UTXO you're trying to exit is more than a week old? Seems unnecessary @colin-axner

**Comment by colin-axner** [comment]
@AdityaSripal In theory no they are not needed. But I could see applications use confirmation signatures to their advantage in practice. For example if you have some contracting service or maybe even rent payment. By transacting on a plasma chain, you can no longer spend that money, only exit it. I could send someone money for like a contract, then send them the confirmation signature upon completion. If I didn't then I would have to exit the utxo and they could use the legal system to show that they are due the amount I am withdrawing. 

I do think verifying confirmation signatures on utxos older than week can be removed, but I think that should be viewed more as a feature or enhancement and should go in a different pr. Preferably one after version 1.0 so developers could decide which feature of the rootchain contract they would prefer.

**Pull request merged**